docs/guides/server/containers.adoc

```diff
@@ -251,4 +252,8 @@ podman|docker run --name mykeycloak -p 8080:8080 \
         start-dev
 ----
 
+WARNING: As the heap size is dynamically calculated based on the total container memory, you should *always set the memory limit* for the container.
+Otherwise, you will see a higher memory consumption as GC does not need to clean up the memory so frequently.
+Once JVM allocates the memory, it is returned to the OS very reluctantly.
```

Comment:
We should consider investigating this more - https://github.com/keycloak/keycloak/issues/28671#issuecomment-2051845838

Based upon https://github.com/keycloak/keycloak/issues/28444#issuecomment-2055586726 this behavior is gargage collector specific and does not line up to most users expectations - even if they set a limit. We may want to consider using ShrinkHeapInSteps to see if forces the default vm to give up heap space more rapidly.

Also relates to #20780

Reply:
I think we might investigate the memory settings for GC, but it seems it's out of the scope of this PR. It might be good to create an issue and gather all the information together there. Or just using an already existing one.